cargo-flash/src/main.rs

```diff
@@ -229,14 +230,68 @@ fn main_try() -> Result<(), failure::Error> {
     let instant = Instant::now();
 
     let mm = session.target.memory_map.clone();
+
+    // Create progress bars.
+    let m =
+        indicatif::MultiProgress::with_draw_target(indicatif::ProgressDrawTarget::stdout_nohz());
+    let style = indicatif::ProgressStyle::default_bar()
+            .tick_chars("⠁⠂⠄⡀⢀⠠⠐⠈✔")
+            .progress_chars("##-")
+            .template("    {msg:.green.bold} {spinner} [{elapsed_precise}] [{wide_bar}] {pos}/{len} (eta {eta})");
+
+    // Create a new progress bar for the erase progress.
+    let pbe = m.add(indicatif::ProgressBar::new(0));
+    pbe.set_style(style.clone());
+    pbe.set_message("Erasing sectors  ");
+
+    // Create a new progress bar for the program progress.
+    let pbp = m.add(indicatif::ProgressBar::new(0));
+    pbp.set_style(style);
+    pbp.set_message("Programming pages");
+
+    // Register callback to update the progress.
+    let progress = FlashProgress::new(move |event| {
+        use ProgressEvent::*;
+        match event {
+            Initialize {
+                total_sectors,
+                total_pages,
+            } => {
+                pbe.set_length(total_sectors as u64);
+                pbp.set_length(total_pages as u64);
+            }
+            PageFlashed { size: _, time: _ } => {
+                pbp.inc(1);
+            }
+            SectorErased { size: _, time: _ } => {
+                pbe.inc(1);
+            }
+            FinishedErasing => {
+                pbe.finish();
+            }
+            FinishedProgramming => {
+                pbp.finish();
+            }
+        }
+    });
+
+    // Make the multi progresses print.
```

Comment:
For me it's not clear why this has to be in a separate thread. Maybe add a comment with an explanation.